src/include/sof/schedule/task.h

```diff
@@ -26,6 +26,11 @@ struct sof;
 #define SOF_TASK_PRI_IPC	SOF_TASK_PRI_HIGH
 #define SOF_TASK_PRI_IDC	SOF_TASK_PRI_HIGH
 
+/* predefined EDF task deadlines */
+#define SOF_TASK_DEADLINE_IDLE		UINT64_MAX
+#define SOF_TASK_DEADLINE_ALMOST_IDLE	(SOF_TASK_DEADLINE_IDLE - 1)
```

Comment:
Hmm, this pops out a bit. "almost-idle" smells of implementing a priority scheduler on top of deadline scheduler. Do we have real need for this?

Reply:
Idle task is the main FW loop, so it should be the last task to be executed. Almost idle is deadline dedicated for tasks, which do not have specific requirements to be executed by the end of deadline, but still should be executed before firmware's idle task.

Reply:
@tlauda Ack. It seems this is used quite widely already before the change, so ok.